pkg/nsx/services/ippool/builder.go

```diff
@@ -0,0 +1,109 @@
+package ippool
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha2"
+	commonctl "github.com/vmware-tanzu/nsx-operator/pkg/controllers/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/util"
+)
+
+var (
+	Int64  = common.Int64
+	String = common.String
+)
+
+func (service *IPPoolService) BuildIPPool(IPPool *v1alpha2.IPPool) (*model.IpAddressPool, []*model.IpAddressPoolBlockSubnet) {
+	return &model.IpAddressPool{
+		Id:          String(service.buildIPPoolID(IPPool)),
+		DisplayName: String(service.buildIPPoolName(IPPool)),
+		Tags:        service.buildIPPoolTags(IPPool),
+	}, service.buildIPSubnets(IPPool)
+}
+
+func (service *IPPoolService) buildIPPoolID(IPPool *v1alpha2.IPPool) string {
+	return strings.Join([]string{"ipc", string(IPPool.UID)}, "_")
```

Comment:
prefix "ipc" stands for?

Reply:
borrow it from ncp

Reply:
ok, could we define a constant for the "ipc" prefix?

Reply:
OK